From dee10102640be4eae1de3949af46a86c5b5f0a0a Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Fri, 23 Feb 2018 17:18:18 +0200
Subject: [PATCH 3422/5242] video: mxsfb: Fix leaking videomem if dispdrv not
 ready

commit  5161adac500697bac8c91306b1da46a20432f944 from
https://source.codeaurora.org/external/imx/linux-imx.git

This is a large leak and repeated probing can even exhaust CMA

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Reviewed-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index ce431b7..57d4de6 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -2285,7 +2285,7 @@ static int mxsfb_probe(struct platform_device *pdev)
 		if (ret == -EPROBE_DEFER)
 			dev_info(&pdev->dev,
 				 "Defer fb probe due to dispdrv not ready\n");
-		goto fb_pm_runtime_disable;
+		goto fb_free_videomem;
 	}
 
 	if (!host->dispdrv) {
-- 
1.7.9.5

